src/packages/version-check/tests/version-check.test.ts

```diff
@@ -0,0 +1,737 @@
+      );
+    });
+    it("process.stdout.columns === -1", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      const oldColumns = process.stdout.columns;
+      process.stdout.columns = -1;
+
+      vc.log();
+      process.stdout.columns = oldColumns;
+      assert.strictEqual(message.length > 0, true, "Message did not log.");
+    });
+    it("process.stdout.columns === null", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      const oldColumns = process.stdout.columns;
+      process.stdout.columns = 0;
+
+      vc.log();
+      process.stdout.columns = oldColumns;
+      assert.strictEqual(message.length > 0, true, "Message did not log.");
+    });
+  });
+
+  describe("getLatestVersion/_fetchLatest", () => {
+    let api;
+    const apiResponse = "1.0.0";
+    const apiSettings = {
+      port: 4000
+    };
+    const sleep = milliseconds => {
+      return new Promise(resolve => setTimeout(resolve, milliseconds));
+    };
+
+    beforeEach(() => {
+      api = http2.createServer();
+      api.on("error", err => console.error(err));
+
+      api.on("stream", async (stream, headers) => {
+        const path = headers[":path"];
+        const method = headers[":method"];
+
+        if (method !== "GET") {
+          stream.respond({
+            ":status": 404
+          });
+          stream.end();
+        }
+        switch (path) {
+          case "/version?package=ganache":
+            if (stream.closed) return;
+            stream.respond({
+              ":status": 200
+            });
+            stream.write(apiResponse);
+            stream.end();
+            break;
+          default:
+            await sleep(10);
+            stream.respond({
+              ":status": 404
+            });
+            stream.end();
+            break;
+        }
+      });
+
+      api.listen(apiSettings.port);
+      vc = new VersionCheck(
+        testVersion,
+        {
+          enabled: true,
+          url: "http://localhost:" + apiSettings.port
+        },
+        testLogger
+      );
+    });
+
+    afterEach(() => {
+      if (api) {
+        api.close();
+      }
+    });
+    it("will not getLatestVersion if version check is disabled", async () => {
+      vc.disable();
+
+      await vc.getLatestVersion();
+
+      const { latestVersion } = vc.getConfig();
+
+      assert.strictEqual(
+        latestVersion,
+        testVersion,
+        "Version Check will _fetchLatest if disabled."
+      );
+    });
+
+    it("fetches the latest version from the API", async () => {
+      let latestVersion;
+
+      latestVersion = await vc._fetchLatest();
+
+      assert.strictEqual(
+        latestVersion === apiResponse,
+        true,
+        "latestVersion does not match apiResponse."
+      );
+    });
+    /*
+    This will only ever cause intermittent failures in mocha on mac.
+    it("When the ttl expires, throw and cancel the request", async () => {
```

Comment:
I assume this is something to do with the test depending on 1ms elapsing ? I don't understand why it's mac specific though.

It'd be nice if we could inject a time provider or something, so that we didn't have that reliance.

Either way, what's the action here? Shall we fix the test, or remove it?

Reply:
I don't know what you mean by time provider but open to learning. 

The failure is `done called multiple times.` You're right about the time elapsing. The test completes before the server finishes so on a normal OS the server/listeners are garbage collected but for whatever reason the PhDs at Apple know better and it seems as though the listeners stick around and when they complete have some implicit `done` invocation.

^^ Massive speculation because I do not have a PhD.

Older versions of mocha would just let you manually call `done` but now you can't if you return a promise. 

I also don't know that it's really our responsibility to test the timeout function so much as we are responsible for the code it executes -that's covered in another test (`destroy`)

Reply:
Spent way too much time trying to understand the interaction that's causing this to be intermittent, with minimal success :(

At least this error is being swallowed, so 🤷

Reply:
Todo: Create issue for the mac test